src/Apps/Sell/Routes/__tests__/PurchaseHistoryRoute.jest.tsx

```diff
@@ -0,0 +1,177 @@
+    router: {
+      push: mockPush,
+      replace: mockReplace,
+    },
+    match: {
+      location: { pathname: "submissions/submission-id/purchase-history" },
+    },
+  }))
+
+  submitMutation = jest.fn(() => ({ catch: () => {} }))
+  ;(useMutation as jest.Mock).mockImplementation(() => {
+    return { submitMutation }
+  })
+})
+
+const { renderWithRelay } = setupTestWrapperTL({
+  Component: (props: any) => {
+    return (
+      <SubmissionRoute submission={props.submission}>
+        <PurchaseHistoryRoute submission={props.submission} />
+      </SubmissionRoute>
+    )
+  },
+  query: graphql`
+    query PurchaseHistoryRoute_Test_Query @raw_response_type {
+      submission(id: "submission-id") {
+        ...SubmissionRoute_submission
+        ...PurchaseHistoryRoute_submission
+      }
+    }
+  `,
+})
+
+describe("PurchaseHistoryRoute", () => {
+  it("renders the Purchase History step", () => {
+    renderWithRelay({
+      ConsignmentSubmission: () => submissionMock2,
+    })
+
+    expect(
+      screen.getByText("Where did you purchase the artwork?")
+    ).toBeInTheDocument()
+    expect(screen.getByText("Back")).toBeInTheDocument()
+    expect(screen.getByText("Continue")).toBeInTheDocument()
+    expect(screen.getByText("Save & Exit")).toBeInTheDocument()
+  })
+
+  it("initially displayes the form epty", async () => {
+    renderWithRelay({
+      ConsignmentSubmission: () => {},
+    })
+
+    await waitFor(() => {
+      expect(screen.getByTestId("provenance-input")).toHaveValue(undefined)
+      expect(screen.getByTestId("signature-radio-no")).not.toBeChecked()
+      expect(screen.getByTestId("signature-radio-yes")).not.toBeChecked()
+    })
+  })
+
+  it("initializes the form with the submission data", async () => {
+    renderWithRelay({
+      ConsignmentSubmission: () => submissionMock,
+    })
+
+    await waitFor(() => {
+      expect(screen.getByTestId("provenance-input")).toHaveValue("Other")
+      expect(screen.getByTestId("signature-radio-no")).toBeChecked()
+      expect(screen.getByTestId("signature-radio-yes")).not.toBeChecked()
+    })
+  })
+
+  it("saves the submission & navigates to the next step when Continue button is clicked", async () => {
+    renderWithRelay({
+      ConsignmentSubmission: () => submissionMock,
+    })
+
+    screen.getByText("Continue").click()
+
+    await waitFor(() => {
+      expect(mockPush).toHaveBeenCalledWith(
+        '/sell2/submissions/<mock-value-for-field-"externalId">/dimensions'
+      )
+
+      expect(submitMutation).toHaveBeenCalledWith(
+        expect.objectContaining({
+          variables: {
+            input: {
+              externalId: '<mock-value-for-field-"externalId">',
+              provenance: "Other",
+              signature: false,
+            },
+          },
+        })
+      )
+    })
+  })
+
+  it("does not save the submission & navigates to the previous step when the back button is clicked", async () => {
```

Comment:
This test we might need to update because now clicking on "Back" will save the submission as well. Rebasing should make this test fail.

Reply:
Oki, I will update it, thanks!